arangod/RocksDBEngine/RocksDBReplicationContext.cpp

```diff
@@ -769,109 +772,165 @@ void RocksDBReplicationContext::extendLifetime(double ttl) {
   _expires = TRI_microtime() + ttl;
 }
 
-void RocksDBReplicationContext::releaseDumpingResources() {
-  if (_trx != nullptr) {
-    _trx->abort();
-    _trx.reset();
-  }
-  if (_collection) {
-    _collection->release();
-    _collection = nullptr;
+/// create snapshot, locks mutex
```

Comment:
Inaccurate comment. Should be "requires mutex to already be locked by this thread" or something.